### PR TITLE
SpotBugsHtmlReport: getting file from path should happen in getter 

### LINKED ISSUE #33: Applying resolution strategies to all configurations does not work when spotbugsMain task is configured
Hi @KengoTODA ,

I found that when you have a build file like this:

```
buildscript {
    repositories {
        maven {
            url "https://plugins.gradle.org/m2/"
        }
    }
    dependencies {
        classpath "gradle.plugin.com.github.spotbugs.snom:spotbugs-gradle-plugin-v2:0.4.4"
    }
}

plugins {
    id 'java'
}

repositories {
    jcenter()
}
allprojects {
    apply plugin: "jp.skypencil.spotbugs.snom"
    
    spotbugsMain {
        reports {
            html {
                enabled = true
                destination = file("$buildDir/reports/spotbugs/main/spotbugs.html")
                stylesheet = 'fancy-hist.xsl'
            }
        }
    }

    configurations.all {
        exclude group: 'ch.qos.logback', module: 'logback-classic'
        exclude group: 'ch.qos.logback', module: 'logback-core'
        exclude group: 'org.apache.logging.log4j', module: 'log4j-to-slf4j'
        exclude group: 'org.slf4j', module: 'slf4j-log4j12'
    }

}
```

When running `./gradlew spotbugsMain` you would get the following error:

```
Caused by: org.gradle.api.InvalidUserDataException: Cannot change dependencies of configuration ':spotbugs' after it has been resolved.
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.preventIllegalMutation(DefaultConfiguration.java:1147)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.validateMutation(DefaultConfiguration.java:1118)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.exclude(DefaultConfiguration.java:906)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.exclude(DefaultConfiguration.java:140)
        at org.gradle.internal.metaobject.BeanDynamicObject$MetaClassAdapter.invokeMethod(BeanDynamicObject.java:483)
        at org.gradle.internal.metaobject.BeanDynamicObject.tryInvokeMethod(BeanDynamicObject.java:195)
        at org.gradle.internal.metaobject.CompositeDynamicObject.tryInvokeMethod(CompositeDynamicObject.java:98)
        at org.gradle.internal.extensibility.MixInClosurePropertiesAsMethodsDynamicObject.tryInvokeMethod(MixInClosurePropertiesAsMethodsDynamicObject.java:34)
        at org.gradle.internal.metaobject.ConfigureDelegate.invokeMethod(ConfigureDelegate.java:56)
```

while re-ordering the `configurations` code block to be before `spotbugsMain` works, one use case is to have a multi-module project where the root build file applies spotbugs to all and then one of the subprojects do changes to the resolution strategies.

This works with current plugin `apply plugin: "com.github.spotbugs"`

Also it seems that the problem is only with `stylesheet`

### PR DESCRIPTION
**Pull request by rpalcolea**
`SpotBugsHtmlReport`: getting file from path should happen in getter  instead of setter to prevent resolving a configuration and slow/break a build that has opinions in resolution strategy. 

This should fix https://github.com/KengoTODA/spotbugs-gradle-plugin-v2/issues/33

### CODE REVIEW

**Comment by rpalcolea** [comment]
Build failed becase of sonar:

```
> Task :sonarqube FAILED
21 actionable tasks: 21 executed
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':sonarqube'.
> You're not authorized to run analysis. Please contact the project administrator.

```

File: src/main/groovy/com/github/spotbugs/snom/internal/SpotBugsHtmlReport.java [hunk 1]
```diff
@@ -53,33 +63,45 @@ public String getName() {
...
             .stream()
             .findFirst();
-    if (spotbugsJar.isPresent()) {
-      TextResource textResource =
-          getTask().getProject().getResources().getText().fromArchiveEntry(spotbugsJar.get(), path);
-      setStylesheet(textResource);
+    if (spotbugs.isPresent()) {
+      File jar = spotbugs.get();
+      logger.debug(
+          "Specified stylesheet ({}) found in spotbugs configuration: {}",
```
Comment (KengoTODA) [inline src/main/groovy/com/github/spotbugs/snom/internal/SpotBugsHtmlReport.java:86]: At this moment we're not sure that the spotbugs jar file contains specified stylesheet or not, right?
I feel this log message is not so intuitive.

**Review by KengoTODA — CHANGES_REQUESTED** [review]

**Comment by KengoTODA** [comment]
I've disabled SonarQube since it's under maintenance. Merge the latest `origin/master` then the build should pass.

**Comment by KengoTODA** [comment]
Created #45 with a functional test to reproduce.

**Pull request closed without merging**